service/storage_service.cc

```diff
@@ -2575,6 +2578,9 @@ future<node_ops_cmd_response> storage_service::node_ops_cmd_handler(gms::inet_ad
             node_ops_update_heartbeat(ops_uuid).get();
         } else if (req.cmd == node_ops_cmd::removenode_done) {
             slogger.info("removenode[{}]: Marked ops done from coordinator={}", req.ops_uuid, coordinator);
+            if (req.fencing_token) {
+                _rpc_fence.increment(*req.fencing_token).get();
+            }
```

Comment:
What happens if a node is down when this happens? Does it get excluded?

Reply:
Ah, maybe the logic is
 - the fencing token is propagated via group 0
 - a down node will reconnect any synchronize its fencing token
 - it will then start to work

But I think that this is to narrow. We don't want to just exclude removed nodes, we want to exclude in-flight RPCs that read/write stale replicas. For that a simple version is unsufficient, because a replica can be valid for multiple topology transitions.

I don't think we want separate fencing tokens for node removals and replica movements, they are too similar.

Reply:
Yes, raft topology will increment the epoch each time replica set for some tokens changes for whatever reason. We do not need `rpc_fence`. Raft will maintain the epoch together with the topology.